### PR TITLE
Add build_resolvers package

### PR DESCRIPTION
**Pull request by natebosch**
This will allow us to drop the dependency on `build_barback`, as well as
give us a place to put future enhancements for Resolvers.

- Copy relevant code from `code_transformers` and replace barback
imports with build imports
- Drop dependency on `build_barback` from `build_runner`
- Use the new resolver in `build_runner`
- Tweak some logging and remove usage of `package:source_span`

### CODE REVIEW

**Comment by natebosch** [comment]
I don't see any significant performance improvement with this - we had hoped there might be some with less translation between `AssetId` representations etc.

This is a hacky copy - we can us it to drive future improvements and once we can start removing other places we use `build_barback` we'll start to see pruned dependencies as well.

**Comment by natebosch** [comment]
I'm fine holding this until after we've published `build_runner` as it is now.

Will take a look at bringing over the tests

**Comment by natebosch** [comment]
Ported over the tests and modified them to use the `resolveSource` stuff from `build_test`. Deleted the ones that had to do with handling file updates.

File: build_resolvers/lib/src/resolver.dart [hunk 1]
```diff
@@ -195,8 +195,11 @@ class AnalyzerResolver implements ReleasableResolver {
 
   @override
   Future<LibraryElement> findLibraryByName(String libraryName) async =>
-      libraries.firstWhere((l) => l.name == libraryName,
-          defaultValue: () => null) as LibraryElement;
+      (await libraries.firstWhere((l) {
+        print('Name: ${l.name}');
```
Comment (jakemac53) [inline build_resolvers/lib/src/resolver.dart]: probably want to remove this :)

File: build_resolvers/test/resolver_test.dart [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+              main() {
+              } ''',
+      }, (resolver) async {
+        var lib = await resolver.libraryFor(entryPoint);
+        expect(lib.importedLibraries.length, 2);
+      }, resolvers: new AnalyzerResolvers());
+    });
+
+    //test('should list all libraries', () {
```
Comment (jakemac53) [inline build_resolvers/test/resolver_test.dart]: is this still failing or just accidentally commented out?

Comment (natebosch) [reply build_resolvers/test/resolver_test.dart]: Just a bad rebase after we got the tests working.

File: build_runner/lib/src/generate/fold_frames.dart [hunk 3]
```diff
@@ -8,6 +8,7 @@ const _foldFramesFor = const [
   '_bazel_codegen',
   'build',
   'build_barback',
```
Comment (kevmoo) [inline build_runner/lib/src/generate/fold_frames.dart]: Remove this?

Comment (natebosch) [reply build_runner/lib/src/generate/fold_frames.dart]: if it showed up (you can still pass the `BarbackResolvers` through if you want) we'd want it folded. We should remove this if and when we delete the package.

**Pull request merged**